backend/cfg/cfgize.ml

```diff
@@ -620,6 +583,90 @@ let update_trap_handler_blocks : State.t -> Cfg.t -> unit =
       | Some block -> block.is_trap_handler <- true)
     (State.get_exception_handlers state)
 
+module Trap_depth_and_exns = struct
+  type handler_stack = Label.t list
+
+  type handler_table = handler_stack Label.Tbl.t
+
+  let record_handler :
+      type a.
+      handler_stack -> handler_table -> can_raise:(a -> bool) -> a -> unit =
+   fun stack table ~can_raise instr ->
+    if can_raise instr
+    then
+      match stack with
+      | [] -> ()
+      | handler_label :: handler_stack ->
+        Label.Tbl.replace table handler_label handler_stack
```

Comment:
It may be worth refactoring the table (if it's not too much work) to check that the same stack is always associated with a given trap handler label.